src/Uno.UI/UI/Xaml/UIElement.EventRegistration.wasm.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Uno;
+using Uno.Extensions;
+using Uno.Logging;
+using Uno.UI.Xaml;
+
+namespace Windows.UI.Xaml
+{
+	partial class UIElement
+	{
+		private class EventRegistration
+		{
+			private static readonly string[] noRegistrationEventNames = { "loading", "loaded", "unloaded" };
+
+			private readonly UIElement _owner;
+			private readonly string _eventName;
+			private readonly bool _canBubbleNatively;
+			private readonly EventArgsParser _payloadConverter;
+			private readonly Action _subscribeCommand;
+
+			private List<Delegate> _invocationList = new List<Delegate>();
+			private List<Delegate> _pendingInvocationList;
+			private bool _isSubscribed = false;
+			private bool _isDispatching;
+
+			public EventRegistration(
+				UIElement owner,
+				string eventName,
+				bool onCapturePhase = false,
+				bool canBubbleNatively = false,
+				HtmlEventFilter? eventFilter = null,
+				HtmlEventExtractor? eventExtractor = null,
+				EventArgsParser payloadConverter = null)
+			{
+				_owner = owner;
+				_eventName = eventName;
+				_canBubbleNatively = canBubbleNatively;
+				_payloadConverter = payloadConverter;
+				if (noRegistrationEventNames.Contains(eventName))
+				{
+					_subscribeCommand = null;
+				}
+				else
+				{
+					_subscribeCommand = () => WindowManagerInterop.RegisterEventOnView(_owner.HtmlId, eventName, onCapturePhase, eventFilter?.ToString(), eventExtractor?.ToString());
+				}
+			}
+
+			public void Add(Delegate handler)
+			{
+				// Do not alter the invocation list while enumerating it (_isDispatching)
+				var invocationList = _isDispatching
+					? _pendingInvocationList ?? (_pendingInvocationList = new List<Delegate>(_invocationList))
+					: _invocationList;
+
+				if (invocationList.Contains(handler))
+				{
+					return;
+				}
+
+				invocationList.Add(handler);
+				if (_subscribeCommand != null && invocationList.Count == 1 && !_isSubscribed)
+				{
+					_subscribeCommand();
+					_isSubscribed = true;
+				}
+			}
+
+			public void Remove(Delegate handler)
+			{
+				// Do not alter the invocation list while enumerating it (_isDispatching)
+				var invocationList = _isDispatching
+					? _pendingInvocationList ?? (_pendingInvocationList = new List<Delegate>(_invocationList))
+					: _invocationList;
+
+				invocationList.Remove(handler);
+
+				// TODO: Removing handler in HTML not supported yet
+				// var command = $"Uno.UI.WindowManager.current.unregisterEventOnView(\"{HtmlId}\", \"{eventName}\");";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826507)